airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -417,12 +421,8 @@ public void connectionUpdated() {
 
   @Override
   public void resetConnection() {
-    workflowState.setResetConnection(true);
-    workflowState.setResetWithScheduling(false);
-    if (workflowState.isRunning()) {
-      workflowState.setCancelledForReset(true);
-      cancellableSyncWorkflow.cancel();
-    }
+    workflowState.setCancelledForReset(true);
+    cancellableSyncWorkflow.cancel();
```

Comment:
We now cancel the workflow whenever resetConnection is called, so that we are executing the same code path every time.